test/bootloader_test.rb

```diff
@@ -0,0 +1,146 @@
+require_relative "test_helper"
+
+Yast.import "Bootloader"
+
+describe Yast::Bootloader do
+  subject { Yast::Bootloader }
+
+  describe ".Import" do
+    # FIXME: looks useless as import is used in autoinstallation and in such case reset do nothing
+    it "resets configuration" do
+      expect(subject).to receive(:Reset)
+
+      subject.Import({})
+    end
+
+    it "marks that configuration is read" do
+      subject.Import({})
+
+      expect(Yast::BootCommon.was_read).to eq true
+    end
+
+    it "marks that configuration is already proposed" do
+      subject.Import({})
+
+      expect(Yast::BootCommon.was_proposed).to eq true
+    end
+
+    it "marks that configuration is changed" do
+      subject.Import({})
+
+      expect(Yast::BootCommon.changed).to eq true
+    end
+
+    it "marks that stage1 location changed" do
+      subject.Import({})
+
+      expect(Yast::BootCommon.location_changed).to eq true
+    end
+
+    it "sets bootloader from key \"loader_type\"" do
+      expect(Yast::BootCommon).to receive(:setLoaderType).with("grub2")
+      subject.Import("loader_type" => "grub2")
+    end
+
+    it "sets proposed bootloader if not set in data" do
+      expect(Yast::BootCommon).to receive(:setLoaderType).with(Yast::BootCommon.getLoaderType(true))
+      subject.Import({})
+    end
+
+    it "acts like missing if \"loader_type\" value is empty" do
+      expect(Yast::BootCommon).to receive(:setLoaderType).with(Yast::BootCommon.getLoaderType(true))
+      subject.Import("loader_type" => "")
+    end
+
+    it "pass initrd specific map to initrd module" do
+      initrd_map = {
```

Comment:
I want to use some real one, with real values, as it is not trivial to find how it should look like :)